Replace vendored osmosis-osm-binary code with Maven hosted equivalent

brettch opened the pull request:
This removes the `osmosis-osm-binary` project containing vendored PBF code with a direct dependency on the `org.openstreetmap.pbf:osmpbf:1.5.0` module from Maven central.

simonpoole commented:
Don't know if you think this warrants a note somewhere in the docs, but as noted here https://github.com/openstreetmap/osmosis/issues/85#issuecomment-912494956 this breaks compatibility with Android 7 and earlier.

brettch commented:
I must admit I'd completely forgotten about that discussion.  Any suggestions on where it should go?  The next release notes perhaps?

simonpoole commented:
> ... The next release notes perhaps?

That's probably best.

brettch commented:
Done.  I've added a "Pending changes" section to the release notes containing the changes since last version and included a note about breaking Android 7 compatibility.  Let me know if it makes sense.  Feel free to send through some improved wording.
